src/dataworker/Dataworker.ts

```diff
@@ -1874,10 +1874,41 @@ export class Dataworker {
         return;
       }
 
+      // Check how liquidReserves will be affected by the exchange rate update and skip it if it wouldn't increase.
+      // Updating exchange rate current or sync-ing pooled tokens is used only to potentially increase liquid
+      // reserves available to the HubPool to execute pool rebalance leaves, particularly fot tokens that haven't
+      // updated recently. If the liquid reserves would not increase, then we skip the update.
+      const hubPool = this.clients.hubPoolClient.hubPool;
+      const multicallInput = [
+        hubPool.interface.encodeFunctionData("pooledTokens", [l1Token]),
+        hubPool.interface.encodeFunctionData("sync", [l1Token]),
+        hubPool.interface.encodeFunctionData("pooledTokens", [l1Token]),
+      ];
+      const multicallOutput = await hubPool.callStatic.multicall(multicallInput);
+      const currentPooledTokens = hubPool.interface.decodeFunctionResult("pooledTokens", multicallOutput[0]);
+      const updatedPooledTokens = hubPool.interface.decodeFunctionResult("pooledTokens", multicallOutput[2]);
+      const liquidReservesDelta = updatedPooledTokens.liquidReserves.sub(currentPooledTokens.liquidReserves);
+
+      // If the delta is positive, then the update will increase liquid reserves and
+      // at this point, we want to update the liquid reserves to make more available
+      // for executing a pool rebalance leaf.
+      const chainId = this.clients.hubPoolClient.chainId;
+      const tokenSymbol = this.clients.hubPoolClient.getTokenInfo(chainId, l1Token)?.symbol;
+
+      if (liquidReservesDelta.lte(0)) {
+        this.logger.debug({
+          at: "Dataworker#_updateExchangeRates",
+          message: `Skipping exchange rate update for l1 token ${tokenSymbol} because liquid reserves would not increase`,
```

Comment:
We use the symbol, so the chain isn't really relevant (and is anyway implied by the mention of an exchange rate).

```suggestion
          message: `Skipping exchange rate update for ${tokenSymbol} because liquid reserves would not increase`,
```